pulp_rpm/app/viewsets.py

```diff
@@ -994,6 +1021,22 @@ class CompsXmlViewSet(viewsets.ViewSet):
     ViewSet for comps.xml Upload.
     """
 
+    # TODO: adjust this policy after upload access policy design done and in place
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list", "retrieve"],
```

Comment:
Did you look at the Comps viewset? It does not seem to have allowed any of those actions, except for POST
```
$ http https://pulp3-source-fedora34.fluffy.example.com/pulp/api/v3/rpm/comps/
HTTP/1.1 405 Method Not Allowed
Access-Control-Expose-Headers: Correlation-ID
Allow: POST, OPTIONS
Connection: keep-alive
Content-Length: 40
Content-Type: application/json
Correlation-ID: 23f525f09c8b44b483830c58bb6a00c0
Date: Thu, 14 Apr 2022 11:24:24 GMT
Referrer-Policy: same-origin
Server: nginx/1.20.1
Vary: Accept, Cookie
X-Content-Type-Options: nosniff
X-Frame-Options: DENY

{
    "detail": "Method \"GET\" not allowed."
}

```